ngraph/core/include/ngraph/op/if.hpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+
+#include "ngraph/function.hpp"
+#include "ngraph/op/parameter.hpp"
+#include "ngraph/op/util/multi_subgraph_base.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v7
+        {
+            /// \brief  If operation.
+            class NGRAPH_API If : public util::MultiSubGraphOp
+            {
+            public:
+                enum BodyIndexes
+                {
+                    then_body_index = 0,
+                    else_body_index = 1
+                };
+
+                NGRAPH_RTTI_DECLARATION;
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                /// \brief     Constructs If with condition
+                ///
+                /// \param     execution_condition   condition node.
+                If(const Output<Node>& execution_condition);
+                If();
+                explicit If(const OutputVector& values);
+
+                std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+                /// \return then_body as ngraph::Function.
+                std::shared_ptr<Function> get_then_body() const
+                {
+                    return m_bodies[then_body_index];
+                }
+                /// \return else_body as ngraph::Function.
+                std::shared_ptr<Function> get_else_body() const
+                {
+                    return m_bodies[else_body_index];
+                }
+                /// \brief     sets new ngraph::Function as new then_body.
+                ///
+                /// \param     body   new body for 'then' branch.
+                void set_then_body(const std::shared_ptr<Function>& body)
+                {
+                    m_bodies[then_body_index] = body;
+                }
+                /// \brief     sets new ngraph::Function as new else_body.
+                ///
+                /// \param     body   new body for 'else' branch.
+                void set_else_body(const std::shared_ptr<Function>& body)
+                {
+                    m_bodies[else_body_index] = body;
+                }
```

Comment:
Why aren't `then` and `else` bodies set through constructor? I hardly imagine that after `If` node creation the bodies can be changed.